### PR TITLE
aggregations

### PR DESCRIPTION
**Pull request by mustafaiman**
Adds fast aggregations + tests + API Doc
Code sample will follow up.

### CODE REVIEW

**Comment by mustafaiman** [comment]
verify

**Comment by mustafaiman** [comment]
verify

File: src/aggregation/AggregatorFactory.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    private logger = LoggingService.getLoggingService();
+    static readonly FACTORY_ID = -41;
+
+    static readonly BIG_DECIMAL_AVG = 0; //not implemented in node.js
+    static readonly BIG_DECIMAL_SUM = 1; //not implemented in node.js
+    static readonly BIG_INT_AVG = 2; //not implemented in node.js
+    static readonly BIG_INT_SUM = 3; //not implemented in node.js
+    static readonly COUNT = 4;
+    static readonly DISTINCT = 5; //returns java serializable, not usable in node.js
```
Comment (sancar) [inline src/aggregation/AggregatorFactory.ts]: Was this because serialization can't handle Set?

Comment (mustafaiman) [reply src/aggregation/AggregatorFactory.ts]: `DistinctValuesAggregator.aggregate()` on Java side returns `Set`. Therefore client receives a Java `Set` as response from the server.

**Pull request merged**